File: app/src/main/java/ru/aradxxx/ciceronetabssample/MainActivity.java
package ru.aradxxx.ciceronetabssample;

import android.os.Bundle;

import ru.aradxxx.ciceronetabs.NavigationContainer;
import ru.aradxxx.ciceronetabs.TabCicerone;
import ru.aradxxx.ciceronetabs.TabNavigator;
import ru.aradxxx.ciceronetabs.TabRouter;
import ru.aradxxx.ciceronetabssample.R;

import ru.aradxxx.ciceronetabssample.tabfragment.Tab1Screen;
import ru.aradxxx.ciceronetabssample.tabfragment.Tab2Screen;
import ru.aradxxx.ciceronetabssample.tabfragment.Tab3Screen;
import ru.aradxxx.ciceronetabssample.tabfragment.TabContainerScreen;

import javax.inject.Inject;

import androidx.annotation.NonNull;
import dagger.android.support.DaggerAppCompatActivity;

/**
 * Пример активити приложения
 */
public class MainActivity extends DaggerAppCompatActivity
        implements NavigationContainer<TabRouter> {
    @Inject
    TabCicerone<TabRouter> tabCicerone;
    private TabNavigator<TabRouter> navigator;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        navigator = new TabNavigator<>(this, tabCicerone, R.id.container);

        if (savedInstanceState == null) {
            firstNavigation();
        }
    }

    @Override
    protected void onResumeFragments() {
        super.onResumeFragments();
        tabCicerone.activityCicerone().getNavigatorHolder().setNavigator(navigator);
    }

    @Override
    protected void onPause() {
        tabCicerone.activityCicerone().getNavigatorHolder().removeNavigator();
        super.onPause();
    }

    @NonNull
    @Override
    public TabRouter router() {
        return tabCicerone.activityRouter();
    }

    private void firstNavigation() {
        router().newRootScreen(new TabContainerScreen());
        router().switchTab(new Tab2Screen());
        router().switchTab(new Tab3Screen());
        router().switchTab(new Tab1Screen());
        router().newRootScreen(Tab.Tab1.tag(), new AppFragmentScreen("FROM ACTIVITY"));
    }
}

File: app/src/main/java/com/github/aradxxx/ciceroneflowssample/MainActivity.kt
package com.github.aradxxx.ciceroneflowssample

import android.os.Bundle
import com.github.aradxxx.ciceroneflow.FlowCicerone
import com.github.aradxxx.ciceroneflow.FlowNavigator
import com.github.aradxxx.ciceroneflow.FlowRouter
import com.github.aradxxx.ciceroneflow.NavigationContainer
import com.github.aradxxx.ciceroneflowssample.tabfragment.tab1Screen
import com.github.aradxxx.ciceroneflowssample.tabfragment.tabContainerScreen
import dagger.android.support.DaggerAppCompatActivity
import javax.inject.Inject

/**
 * Пример активити приложения
 */
class MainActivity : DaggerAppCompatActivity(), NavigationContainer<FlowRouter> {
    @Inject
    lateinit var flowCicerone: FlowCicerone<FlowRouter>
    private val navigator: FlowNavigator<FlowRouter> by lazy {
        FlowNavigator(
            this,
            R.id.container,
            flowCicerone)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (savedInstanceState == null) {
            firstNavigation()
        }
    }

    override fun onResumeFragments() {
        super.onResumeFragments()
        flowCicerone.mainCicerone().getNavigatorHolder().setNavigator(navigator)
    }

    override fun onPause() {
        flowCicerone.mainCicerone().getNavigatorHolder().removeNavigator()
        super.onPause()
    }

    override fun router(): FlowRouter {
        return flowCicerone.mainRouter()
    }

    private fun firstNavigation() {
        router().newRootScreen(tabContainerScreen())
        router().newRootScreen(Tab.Tab1.tag(), appFragmentScreen("FROM ACTIVITY 1"))
        router().newRootScreen(Tab.Tab3.tag(), appFragmentScreen("FROM ACTIVITY 3"))
        router().newRootScreen(Tab.Tab2.tag(), appFragmentScreen("FROM ACTIVITY 2"))
        router().switch(tab1Screen())
    }
}
